bittensor/axon.py

```diff
@@ -151,6 +151,61 @@ def stop(self):
             self.should_exit = True
 
 
+def load_nonces(
+    nonces_basepath: str, coldkey: str, hotkey: str, ip: str, port: int
+) -> Dict[str, int]:
+    """
+    Loads the nonces from the specified file path using hash of coldkey:hotkey:netuid and returns them as a dictionary.
+
+    Args:
+        nonces_path (str): The file path to load the nonces from.
+
+    Returns:
+        Dict[str, int]: A dictionary containing the (endpoint_key, nonce) pairs loaded from the file.
+    """
+    if not os.path.exists(nonces_basepath):
+        os.makedirs(nonces_basepath)
+
+    nonce_dirname = str(hash(f"{coldkey}:{hotkey}:{ip}:{port}"))
+    nonces_path = os.path.join(nonces_basepath, nonce_dirname, "nonces.json")
+
+    if os.path.exists(nonces_path):
+        with open(nonces_path, "r") as f:
+            nonces = json.load(f)
+        return nonces
+
+    return {}
+
+
+def save_nonces(
+    nonces_basepath: str,
+    coldkey: str,
+    hotkey: str,
+    ip: str,
+    port: int,
+    nonces: Dict[str, int],
+):
+    """
+    Saves the nonces to the specified file path using hash of coldkey:hotkey:netuid.
+
+    Args:
+        nonces_path (str): The file path to save the nonces to.
+        coldkey (str): The coldkey to use in the hash.
+        hotkey (str): The hotkey to use in the hash.
+        ip (str): The ip to use in the hash.
+        port (int): The port to use in the hash.
+        nonces (Dict[str, int]): A dictionary containing the (endpoint_key, nonce) pairs to save to the file.
+    """
+    if not os.path.exists(nonces_basepath):
+        os.makedirs(nonces_basepath)
```

Comment:
```suggestion
    os.makedirs(nonces_basepath, exists_ok=True)
```
less code and one less race condition